src/acl/acl.ts

```diff
@@ -170,16 +167,16 @@ export function hasResourceAcl<Resource extends WithAcl & WithResourceInfo>(
 }
 
 /**
- * Access the ACL attached to a Resource.
+ * Return the resource ACL attached to a Resource.
  *
- * Given a Resource that has an ACL attached, this function will give you access to that ACL. To
- * verify whether the ACL is available, see [[hasResourceAcl]].
+ * Given a Resource that has its Resource ACL attached and is accessible by the user
+ * (see [[hasResourceAcl]]), this function returns the Resource ACL.
  *
- * Please note that the Web Access Control specification is not yet finalised, and hence, this
- * function is still experimental and can change in a non-major release.
+ * _Note_: The Web Access Control specification is not yet finalised. As such, this
+ * function is still experimental and subject to change.
  *
  * @param resource A Resource with potentially an ACL attached.
- * @returns The ACL, if available, and undefined if not.
+ * @returns The resource ACL if available and `null` if not.
```

Comment:
Same thing here:

We seem to return either null or the resourceAcl

but we have 
`
export function getResourceACL(): AclDataset
`
so that the typedoc states it returns AclDataset followed by the above sententce that this returns the resource ACL.

Reply:
`AclDataset` is the type of any ACL, so both a resource ACL and a fallback ACL are of type `AclDataset`. I'm not sure that's what you are asking though :)

Reply:
Whoops, yes, the `undefined` should be `null` (that was changed at some point).

There are multiple instances of `export function getResourceAcl` - these are type overloads. The first says that if you pass in a resource of which we know that it has an ACL attached (i.e. `WithResourceAcl` applies), you will definitely get an `AclDataset` and not `null`. However, if we do not know for sure that it also has a Resource ACL, we might still return `null`.

So yes, a description that says it returns the Resource ACL if it is available, and `null` if not` is correct.

Reply:
Cool. Didn't realize that ACL was type AclDataset.